waiter/src/waiter/token.clj

```diff
@@ -516,48 +516,61 @@
                            :log-level :warn}))))
 
       nil
-      (do
+      (let [existing-editor (get existing-token-metadata "editor")
+            existing-owner (get existing-token-metadata "owner")
+            owner? (and existing-owner (authz/manage-token? entitlement-manager authenticated-user token existing-token-metadata))
+            editor? (and existing-editor (not owner?)  (authz/run-as? entitlement-manager authenticated-user existing-editor))
+            delegated-user (if editor?
+                             (do ;; allow editor to run with owner privileges
+                               (log/info "promoting editor to owner privileges while editing token"
+                                         {:editor authenticated-user :owner existing-owner})
+                               existing-owner)
+                             authenticated-user)]
+        (when editor?
+          (let [existing-token-parameters (sd/token->token-parameters kv-store token :include-deleted false)]
+            (doseq [parameter-name ["editor" "owner" "run-as-user"]]
+              (let [existing-value (get existing-token-parameters parameter-name)
+                    new-value (get new-token-data parameter-name)]
+                (when (not= existing-value new-value)
+                  (throw (ex-info (str "Not allowed to edit parameter " parameter-name)
+                                  {:authenticated-user authenticated-user
+                                   :existing-token-description existing-token-parameters
+                                   :parameter parameter-name
+                                   :parameter-exiting-value existing-value
+                                   :parameter-new-value new-value
+                                   :status http-403-forbidden
+                                   :log-level :warn})))))))
         (when (and run-as-user (not= "*" run-as-user))
-          (when-not (authz/run-as? entitlement-manager authenticated-user run-as-user)
+          (when-not (authz/run-as? entitlement-manager delegated-user run-as-user)
```

Comment:
Previously we were checking if the authenticated user could `run-as?`. Now we're checking if the owner can `run-as?` It's not clear why we are changing the original behavior.